src/app/data-explorer/query-builder-wrapper.service.ts

```diff
@@ -65,18 +66,19 @@ export const mapOptionsArrayToModelDataType = (
 @Injectable({
   providedIn: 'root',
 })
-export class QueryBuilderService {
+export class QueryBuilderWrapperService {
   constructor(private profileService: ProfileService) {}
 
   public setupConfig(
```

Comment:
This is quite a long and complicated method. Could you either break it into simpler private methods or add some more descriptive comments throughout please :).

Reply:
Method has been broken down.